src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+/**
+ * @author Rob Frawley 2nd <rmf@src.run>
+ */
+class MemcachedAdapter extends AbstractAdapter
+{
+    private $client;
+
+    public static function isSupported()
+    {
+        return extension_loaded('memcached') && version_compare(phpversion('memcached'), '2.2.0', '>=');
+    }
+
+    public function __construct(\Memcached $client, $namespace = '', $defaultLifetime = 0)
+    {
+        parent::__construct($namespace, $defaultLifetime);
+        $this->client = $client;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doSave(array $values, $lifetime)
+    {
+        return $this->client->setMulti($values, $lifetime)
+            && $this->client->getResultCode() === \Memcached::RES_SUCCESS;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doFetch(array $ids)
+    {
+        return $this->client->getMulti($ids);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doHave($id)
+    {
+        return $this->client->get($id) !== false
+            || $this->client->getResultCode() === \Memcached::RES_SUCCESS;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doDelete(array $ids)
+    {
+        $toDelete = count($ids);
+        foreach ((array) $this->client->deleteMulti($ids) as $result) {
+            if (true === $result || \Memcached::RES_NOTFOUND === $result) {
+                --$toDelete;
+            }
+        }
+
+        return 0 === $toDelete;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doClear($namespace)
+    {
+        if (!isset($namespace[0]) || false === $ids = $this->getIdsByPrefix($namespace)) {
+            return $this->client->flush();
+        }
+
+        $isCleared = true;
+        do {
+            $isCleared = $this->doDelete($ids) && $isCleared;
+        } while ($ids = $this->getIdsByPrefix($namespace));
+
+        return $isCleared;
+    }
+
+    private function getIdsByPrefix($namespace)
```

Comment:
Added a commit with https://github.com/symfony/symfony/pull/20858/files#diff-e01499fab5ad7f5e36517d157bc13e2fR96 to check if any Travis environment actually has a functional combination of the extension and library. After checking https://travis-ci.org/symfony/symfony/builds/183319461, if none actually utilize this, then I'm with you 100%.